sdlib/d/gc/sizeclass.d

```diff
@@ -47,6 +47,69 @@ enum SizeClass {
 
 enum MaxTinySize = ClassCount.Tiny * Quantum;
 
+// Determine whether given size is permitted by allocator.
+bool isAllocatableSize(size_t size) {
+	return size > 0 && size <= MaxAllocationSize;
+}
+
+unittest isAllocatableSize {
+	assert(!isAllocatableSize(0));
+	assert(isAllocatableSize(1));
+	assert(isAllocatableSize(42));
+	assert(isAllocatableSize(99999));
+	assert(isAllocatableSize(MaxAllocationSize));
+	assert(!isAllocatableSize(MaxAllocationSize + 1));
+	assert(!isAllocatableSize(size_t.max));
+}
+
+enum maxSizeClass = ClassCount.Total - 1;
```

Comment:
I'm not sure this necessarily match the max allocatable size (which I think is 16T).

Reply:
There are size you cannot allocated, and they have size classes allocated with them that you cannot allocate either. I'm not sure how this is a problem?

Also, MaxAllocationSize should be closer to 16T than 4G.

Reply:
But also, is that enum even needed?